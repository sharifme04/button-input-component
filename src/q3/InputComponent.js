import React, { useState } from "react";
import PropTypes from "prop-types";

const InputComponent = ({ initialValue }) => {
  const [search, setSearch] = useState(initialValue);
  const onChange = event => setSearch(event.target.value);
  return (
    <div className="col-4">
      <div className="form-group">
        <input
          type="text"
          value={search}
          onChange={onChange}
          className="form-control"
          placeholder="Please enter a text"
          name="search"
        />
      </div>
    </div>
  );
};

InputComponent.propTypes = {
  initialValue: PropTypes.string
};
export default InputComponent;
