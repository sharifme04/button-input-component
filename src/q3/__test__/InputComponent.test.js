import React from "react";
import { shallow } from "enzyme";
import InputComponent from "../InputComponent";

describe("<InputComponent/>", () => {
  test("return Component true", () => {
    const component = shallow(<InputComponent />);
    expect(component).toMatchSnapshot();
  });

  test("return input true", () => {
    const component = shallow(<InputComponent />);
    expect(component.find("input")).toHaveLength(1);
  });
});
