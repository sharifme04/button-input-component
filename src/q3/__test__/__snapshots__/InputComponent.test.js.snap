// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputComponent/> return Component true 1`] = `
<div
  className="col-4"
>
  <div
    className="form-group"
  >
    <input
      className="form-control"
      name="search"
      onChange={[Function]}
      placeholder="Please enter a text"
      type="text"
    />
  </div>
</div>
`;
